### PR TITLE
Branch03 building components and design

### PR DESCRIPTION
**Pull request by MilosJovanovic1986**
**Motivation:** In this branch I'll working on design and also on building a components. For design I'll using `tailwind framework`, if needed I'll use `HTML` and `CSS` as well.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+node_modules/
+*/.next/*
+.next/*
+package-lock.json
```
Comment (masone) [inline .gitignore:4]: please keep this in the project, it helps that everyone installs the exact same versions of the packages

File: frontends/.eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+export default {
+    "env": {
+        "browser": true,
+        "es2021": true
+    },
+    "extends": [
+        "eslint:recommended",
+        "plugin:react/recommended",
+        "plugin:@typescript-eslint/recommended"
```
Comment (masone) [inline frontends/.eslintrc.js]: are you using typescript at all?

File: frontends/.eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    "parserOptions": {
+        "ecmaFeatures": {
+            "jsx": true
+        },
+        "ecmaVersion": 12,
+        "sourceType": "module"
+    },
+    "plugins": [
+        "react",
+        "@typescript-eslint"
```
Comment (masone) [inline frontends/.eslintrc.js]: are you using typescript at all?

File: frontends/.eslintrc.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    ],
+    "rules": {
+    }
+};
```
Comment (masone) [inline frontends/.eslintrc.js:16]: very nice! can you add it to the root folder, so the backend code can be linted the same?

File: frontends/.next/build-manifest.json [hunk 5]
```diff
@@ -0,0 +1,34 @@
+{
```
Comment (masone) [inline frontends/.next/build-manifest.json]: once committed, git does not automatically delete the files when you add them to `.gitignore`. you'll have to git-delete them manually again

File: frontends/package.json [hunk 6]
```diff
@@ -0,0 +1,42 @@
+{
+  "name": "ipa-frontend",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "dev": "next -p 7759c",
```
Comment (masone) [inline frontends/package.json]:
```suggestion
    "dev": "next -p 7759",
```

File: frontends/package.json [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    "build": "next build",
+    "build:css": "postcss src/style/tailwind.css -o src/style/main.css",
+    "build:watch": "postcss src/tailwind.css -o src/main.css --watch",
+    "lint": "npm run eslint -- js,jsx,json",
+    "format": "npm run format:js && npm run format:css",
```
Comment (masone) [inline frontends/package.json]: `format:css` is not defined, so this command will fail

File: frontends/components/commands.js [hunk 8]
```diff
@@ -0,0 +1,112 @@
+import socket from "../socket";
+import { MdFlightLand, MdFlightTakeoff, MdForward } from "react-icons/md";
+import {
+  HiArrowDown,
+  HiArrowLeft,
+  HiArrowRight,
+  HiArrowUp,
+} from "react-icons/hi";
+import { BiArrowToBottom, BiArrowToTop } from "react-icons/bi";
```
Comment (masone) [inline frontends/components/commands.js:9]: some dependencies here are missing in the package.json

File: frontends/package.json (hunk 6 above)
Comment (masone) [reply frontends/package.json]: I'm not sure why, but it works with the `c` too. does it have any meaning I don't know about?

File: frontends/package.json [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    "express": "^4.17.1",
+    "lodash": "^4.17.21",
+    "lodash.isfunction": "^3.0.9",
+    "next": "^10.0.7",
+    "postcss-cli": "^8.3.1",
+    "react": "^17.0.1",
+    "react-dom": "^17.0.1",
+    "react-icons": "^4.2.0",
+    "socket.io-client": "^2.1.1",
+    "styled-components": "^4.1.1"
```
Comment (masone) [inline frontends/package.json]: is this dependency being used?

File: frontends/package.json [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+    "build:watch": "postcss src/tailwind.css -o src/main.css --watch",
+    "lint": "npm run eslint -- js,jsx,json",
+    "format": "npm run format:js && npm run format:css",
+    "format:js": "npm run lint -- --fix",
+    "start": "next start"
+  },
+  "author": "Milos Jovanovic",
+  "license": "ISC",
+  "dependencies": {
+    "babel-plugin-styled-components": "^1.9.2",
```
Comment (masone) [inline frontends/package.json]: is this dependency being used?

File: frontends/package.json [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+    "express": "^4.17.1",
+    "lodash": "^4.17.21",
+    "lodash.isfunction": "^3.0.9",
```
Comment (masone) [inline frontends/package.json:18]: is this dependency being used?

File: frontends/package.json [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+  "main": "index.js",
+  "scripts": {
+    "dev": "next -p 7759c",
+    "build": "next build",
+    "build:css": "postcss src/style/tailwind.css -o src/style/main.css",
+    "build:watch": "postcss src/tailwind.css -o src/main.css --watch",
+    "lint": "npm run eslint -- js,jsx,json",
+    "format": "npm run format:js && npm run format:css",
+    "format:js": "npm run lint -- --fix",
+    "start": "next start"
```
Comment (masone) [inline frontends/package.json:11]: the instructions in the README don't work. I'm getting the following error when running `npm start`:

```
Error: Could not find a production build in the '/Users/cfelder/carforyou/IPA-Projekt/frontends/.next' directory. Try building your app with 'next build' before starting the production server
````

Maybe this is why you committed the `.next` folder initially. It is generated by a `npm run build` and used for production deployments. For a local run, you can just resort to `npm run dev`, also see https://nextjs.org/docs/getting-started#manual-setup

Please update the README so the instructions for starting the drone works

File: frontends/package.json [hunk 13]
```diff
@@ -0,0 +1,42 @@
+{
+  "name": "ipa-frontend",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "dev": "next -p 7759c",
+    "build": "next build",
+    "build:css": "postcss src/style/tailwind.css -o src/style/main.css",
+    "build:watch": "postcss src/tailwind.css -o src/main.css --watch",
```
Comment (masone) [inline frontends/package.json]: the `main.css` file is never consumed, so I believe you can delete this
also, this task is never run

File: frontends/.eslintrc.js (hunk 2 above)
Comment (MilosJovanovic1986) [reply frontends/.eslintrc.js]: no, will removed it...

**Pull request merged**